jujubigdata/handlers.py

```diff
@@ -131,10 +138,19 @@ def install_java(self):
         unitdata.kv().set('java.version', java_version)
 
     def install_hadoop(self):
-        jujuresources.install('hadoop-%s' %
-                              self.cpu_arch,
-                              destination=self.dist_config.path('hadoop'),
-                              skip_top_level=True)
+        hadoop_version = self.dist_config.hadoop_version
+        try:
+            jujuresources.install('hadoop-%s-%s' %
+                                  (hadoop_version,
+                                   self.cpu_arch),
+                                  destination=self.dist_config.path('hadoop'),
+                                  skip_top_level=True)
+        except:
```

Comment:
This shouldn't be a bare `except`.  We should catch the specific error we expect, in case there's an error unrelated to the version-specific resource not being available.